src/traces/sunburst/plot.js

```diff
@@ -148,8 +150,9 @@ function plotOne(gd, cd, element, transitionOpts) {
     // slice path generation fn
     var pathSlice = function(d) { return Lib.pathAnnulus(d.rpx0, d.rpx1, d.x0, d.x1, cx, cy); };
     // slice text translate x/y
-    var transTextX = function(d) { return cx + d.pxmid[0] * d.transform.rCenter + (d.transform.x || 0); };
-    var transTextY = function(d) { return cy + d.pxmid[1] * d.transform.rCenter + (d.transform.y || 0); };
+
+    var getTargetX = function(d) { return cx + (d.pxtxt || d.pxmid)[0] * d.transform.rCenter + (d.transform.x || 0); };
+    var getTargetY = function(d) { return cy + (d.pxtxt || d.pxmid)[1] * d.transform.rCenter + (d.transform.y || 0); };
```

Comment:
What's this new `d.pxtxt` for?

Reply:
When fitting horizontal text in `pie` and `sunburst` the best position may not be the middle point of the arc. So `pxtxt` keeps the position of text if it is moved form the middle.

![uniformtext_pie_8_horizontal](https://user-images.githubusercontent.com/33888540/70750581-a7c0fb00-1cfc-11ea-837c-fadfae1d3ea7.png)

![sunburst](https://user-images.githubusercontent.com/33888540/70750688-db038a00-1cfc-11ea-90e2-0aa7e51b48dd.png)

Reply:
these are really nice... loving the consistent alignment of labels in the sunburst :)

Reply:
(well, semi-consistent!) We should maybe special-case the noon and 6 o'clock positions as 'preferred' if they result in the same size as 12:05 or 6:15, say :)

Reply:
Good call. Done in https://github.com/plotly/plotly.js/pull/4420/commits/1479943d15bb8fa09e7a3fa913ed2f0d576ea345.